config/main.py

```diff
@@ -2349,6 +2349,126 @@ def speed(ctx, interface_name, interface_speed, verbose):
         command += " -vv"
     clicommon.run_command(command, display_cmd=verbose)
 
+#
+# 'autoneg' subcommand
+#
+
+@interface.command()
+@click.pass_context
+@click.argument('interface_name', metavar='<interface_name>', required=True)
+@click.argument('mode', metavar='<mode>', required=True, type=click.Choice(["enabled", "disabled"]))
+@click.option('-v', '--verbose', is_flag=True, help="Enable verbose output")
+def autoneg(ctx, interface_name, mode, verbose):
+    """Set interface auto negotiation mode"""
+    # Get the config_db connector
+    config_db = ctx.obj['config_db']
+
+    if clicommon.get_interface_naming_mode() == "alias":
+        interface_name = interface_alias_to_name(config_db, interface_name)
+        if interface_name is None:
+            ctx.fail("'interface_name' is None!")
```

Comment:
This is following the style of other "config interface" commands.

Reply:
I still suggest changing it. This message is not clear.
The user can provide an invalid interface name which causes `interface_alias_to_name` to return `None`. But he might get confused about this message because he didn't provide an None or empty interface name.

Reply:
I totally agree your suggestion. Probably we can have a separate PR to change this print. If we only change the print in this subcommand, it will be a litter bit weird. How do you think?

Reply:
Agree. Let's keep it as it is for now.